icu4j/main/tests/core/src/com/ibm/icu/dev/test/util/ULocaleTest.java

```diff
@@ -1001,13 +1001,16 @@ public void TestCanonicalization(){
     }
 
     @Test
-    public void TestGetAvailable(){
+    public void TestGetAvailable() {
         ULocale[] locales = ULocale.getAvailableLocales();
-        if(locales.length<10){
-            errln("Did not get the correct result from getAvailableLocales");
+        if (locales.length < 1) {
+            errln("Did not get any results from getAvailableLocales");
         }
-        if(!locales[locales.length-1].getName().equals("zu_ZA")){
-            errln("Did not get the expected result");
+        for (ULocale locale : locales) {
+            int length = locale.getLanguage().length();
+            if (!(length == 0 || length == 2 || length == 3)) {
```

Comment:
When would you have `length == 0`?

Reply:
I had thought that this code in `ICUResourceBundle.createULocaleList()` under some circumstances would generate a `Locale` entry with the empty string as its identifier:

```java
if (locstr.equals("root")) {
    locales[i++] = ULocale.ROOT;
}
```

But as you didn't expect this to happen and it doesn't happen in practice when I run this test, I guess that code won't be reached from this test.